emu_mps/mps_backend_impl.py

```diff
@@ -728,63 +735,72 @@ def progress(self) -> None:
             SwipeDirection.RIGHT_TO_LEFT,
         ), "Unknown Swipe direction"
 
-        if self.swipe_direction == SwipeDirection.LEFT_TO_RIGHT:
-            left_idx, right_idx = idx, idx + 1
-        elif self.swipe_direction == SwipeDirection.RIGHT_TO_LEFT:
-            left_idx, right_idx = idx - 1, idx
-
         orth_center_right = self.swipe_direction == SwipeDirection.LEFT_TO_RIGHT
         new_L, new_R, energy = minimize_energy_pair(
-            state_factors=self.state.factors[left_idx : right_idx + 1],
-            ham_factors=self.hamiltonian.factors[left_idx : right_idx + 1],
+            state_factors=self.state.factors[idx : idx + 2],
+            ham_factors=self.hamiltonian.factors[idx : idx + 2],
             baths=(self.left_baths[-1], self.right_baths[-1]),
             orth_center_right=orth_center_right,
             config=self.config,
             residual_tolerance=self.residual_tolerance,
         )
-        self.state.factors[left_idx], self.state.factors[right_idx] = new_L, new_R
-        self.state.orthogonality_center = right_idx if orth_center_right else left_idx
+        self.state.factors[idx], self.state.factors[idx + 1] = new_L, new_R
+        self.state.orthogonality_center = idx + 1 if orth_center_right else idx
         self.current_energy = energy
 
         # updating baths and orthogonality center
         if self.swipe_direction == SwipeDirection.LEFT_TO_RIGHT:
+            self._left_to_right_update(idx)
+        elif self.swipe_direction == SwipeDirection.RIGHT_TO_LEFT:
+            self._right_to_left_update(idx)
+        else:
+            raise Exception("Did not expect this")
```

Comment:
This part is a bit suspicious, I think it can be removed. Above you have an assert, which does the job 

```
        assert self.swipe_direction in (
            SwipeDirection.LEFT_TO_RIGHT,
            SwipeDirection.RIGHT_TO_LEFT,
        ), "Unknown Swipe direction"
```